hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,294 @@
+       change in the switch.
+       """
+        jun = o_session
+        jun.cfg.load(
+               template_path=j_template, template_vars=var_dict,
+               format='text'
+               )
+
+
+    def _set_commit_config(self, set_command):
+        """ Changes the switch config by using only 'set' method.
+        `set_command`: 'set' method based command string.
+        Note: Some configurations changes require both 'set'
+        and 'jinja_template' method. In those cases use combination of functions
+        `_set_load_config`, `_jinja_load_config` as needed finally commit using
+        `_commit_config`. 
+        """
+
+        jun = self.session()
+        try:
+            with jun:
+                self._set_load_config(jun, set_command)
+                self._commit_config(jun)
+        except ConnectError as err:
+                print("connat connect to device: {0}".format(err))
+                sys.exit(1)
+
+    def _interface_info(self, port):
+        """Fetches latest committed configuration information about `port`"""
+        jun = self.session()
+        try: 
+            with jun:
+                interface_config = InterfaceConfigTables(jun)
+                interface_config.get(interface= port, options = {'database':'committed'})
+                ic_dict = json.loads(interface_config.to_json())
+        except ConnectError as err:
+                print("connat connect to device: {0}".format(err))
+                sys.exit(1)
+
+        return ic_dict
+
+    def _trunk_mode(self, port):
+        """Returns True if the port is in trunk mode else returns False. """
+        port_info = self._interface_info(port)
+        return port_info[port]['trunk_mode']
+
+    def _set_mode(self, port, mode):
+        """ Will set a given port to 
+        trunk mode or access mode based on the `mode` flag.
+        """
+        if mode == "trunk":
+            set_mode = """
+            set interfaces {port} unit 0 family ethernet-switching interface-mode trunk
+        """.format(port=port)
+            self._set_commit_config(set_mode)
+        else: 
+            self._remove_all_vlans_port(port)
+
+        #x = "set interfaces et-0/0/08:0 unit 0 family ethernet-switching interface-mode trunk"
+
+    def _get_native_vlan(self, port):
+        """ Returns the id of the native vlan for the trunked port 
+        if it exists returns nothing otherwise.  
+        """ 
+        port_info = self._interface_info(port)
+        return port_info[port]['native_vlan']
+
+    def _set_native_vlan(self, port, network_id):
+       """Sets native vlan for a trunked port. """
+       set_native_vlan = """
+       set interface {port} native-vlan-id {network_id}
+        """.format(port=port, network_id=network_id)
+       self._set_commit_config(set_native_vlan)
+
+    def _remove_native_vlan(self, port, network_id):
+        """Removes native_vlan tag from port. """
```

Comment:
in case of this switch, does it actually remove the native vlan or set it to some default?

Reply:
The function is changed. review it again and let me know if you have any questions.